apps/server/src/modules/provisioning/strategy/oidc/service/oidc-provisioning.service.ts

```diff
@@ -185,4 +187,43 @@ export class OidcProvisioningService {
 
 		return filteredUsers;
 	}
+
+	async removeUserFromExternalGroups(
+		externalUserId: EntityId,
+		externalGroups: ExternalGroupDto[],
+		systemId: EntityId
+	): Promise<void> {
+		const user: UserDO | null = await this.userService.findByExternalId(externalUserId, systemId);
+
+		if (!user) {
+			throw new NotFoundLoggableException('User', 'externalId', externalUserId);
+		}
+
+		const existingGroupsOfUser: Group[] = await this.groupService.findByUser(user);
```

Comment:
I would prefer "findGroupsByUser"

Reply:
A.findByB seems to be the convention though. I did not see a A.findAByB function anywhere, so i would leave it that way. But feel free to discuss this further :)